File: src/contests/cg/puzzle/java/rubik.java
// 🎮 CodinGame Puzzle - rubik
// https://www.codingame.com/training/medium/rubik

import java.io.*;

class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        long n = Long.parseLong(br.readLine().trim());
        // Visible mini-cubes = all cubes minus the hidden inner cube of side n-2.
        long inner = n >= 2 ? n - 2 : 0;
        System.out.println(n * n * n - inner * inner * inner);
    }
}

File: src/contests/cg/puzzle/ocaml/rubik.ml
(* 🎮 CodinGame Puzzle - rubik *)
(* https://www.codingame.com/training/medium/rubik *)

let () =
  let n = int_of_string (String.trim (read_line ())) in
  (* Visible mini-cubes = all cubes minus the hidden inner cube of side n-2. *)
  let inner = if n >= 2 then n - 2 else 0 in
  Printf.printf "%d\n" (n * n * n - inner * inner * inner)
